examples/custom_load_balancing_policy.rs

```diff
@@ -18,20 +18,20 @@ struct CustomLoadBalancingPolicy {
     fav_datacenter_name: String,
 }
 
-fn with_random_shard(node: NodeRef) -> (NodeRef, Shard) {
+fn with_random_shard(node: NodeRef) -> (NodeRef, Option<Shard>) {
     let nr_shards = node
         .sharder()
         .map(|sharder| sharder.nr_shards.get())
         .unwrap_or(1);
-    (node, thread_rng().gen_range(0..nr_shards) as Shard)
+    (node, Some(thread_rng().gen_range(0..nr_shards) as Shard))
 }
 
 impl LoadBalancingPolicy for CustomLoadBalancingPolicy {
     fn pick<'a>(
         &'a self,
         _info: &'a RoutingInfo,
         cluster: &'a ClusterData,
-    ) -> Option<(NodeRef<'a>, Shard)> {
+    ) -> Option<(NodeRef<'a>, Option<Shard>)> {
```

Comment:
I'm not fully convinced that's a good idea - unless you have a good name for a target with noni`Option` shard.
In context of LBP target would be, as you said, `(NodeRef<'a>, Option<Shard>)`.
In context of `Plan` it would be `(NodeRef<'a>, Shard)`.
I think we'll reduce confusion by directly exposing the types, instead of hiding them behind another alias. The type is not complicated, so it doesn't harm readability imo.

Reply:
OK, let's leave it as-is.